agent/src/main/java/com/thoughtworks/go/agent/service/AgentUpgradeService.java

```diff
@@ -93,6 +93,11 @@ void checkForUpgrade(String agentMd5, String launcherMd5, String agentPluginsMd5
 
     private void validateMd5(String currentMd5, CloseableHttpResponse response, String agentContentMd5Header, String what) {
         final Header md5Header = response.getFirstHeader(agentContentMd5Header);
+        LOGGER.debug("[Agent Upgrade Validate MD5] validating MD5 for {}, currentMD5:{}, responseMD5:{}, headerKey:{}", what, currentMd5, md5Header, agentContentMd5Header);
+        if (md5Header == null) {
+            String message = String.format("[Agent Upgrade Validate MD5] Did not find %s MD5 header in response %s.", agentContentMd5Header, response);
```

Comment:
Use slf4j formatter, not `String.format`

Reply:
Thanks. Fixed!